tests/main/views/test_marketplace.py

```diff
@@ -1378,32 +1434,234 @@ def test_opportunity_status_and_published_date(self):
+            self._get_dos_brief_search_api_aggregations_response_user_research_fixture_data(),
+        ]
+
+        self._data_api_client_patch = mock.patch('app.main.views.marketplace.data_api_client', autospec=True)
+        self._data_api_client = self._data_api_client_patch.start()
+        self._data_api_client.find_frameworks.return_value = {'frameworks': [
+            {
+                'id': 3,
+                'name': "Digital Outcomes and Specialists 2",
+                'slug': "digital-outcomes-and-specialists-2",
+                'framework': "digital-outcomes-and-specialists",
+                'lots': [
+                    {'name': 'Digital outcomes', 'slug': 'digital-outcomes', 'allowsBrief': True}
+                ],
+                'status': 'live',
+            }]
+        }
+
+    def teardown_method(self, method):
+        self._data_api_client_patch.stop()
+        self._view_search_api_client_patch.stop()
+        self._presenters_search_api_client_patch.stop()
+
+    @pytest.mark.parametrize('query_string, content_type',
+                             (('', 'text/html; charset=utf-8'),
+                              ('?live-results=true', 'application/json')))
+    def test_endpoint_switches_on_live_results_request(self, query_string, content_type):
+        res = self.client.get('/digital-outcomes-and-specialists/opportunities{}'.format(query_string))
+        assert res.status_code == 200
+        assert res.content_type == content_type
+
+    def test_live_results_returns_valid_json_structure(self):
+        res = self.client.get('/digital-outcomes-and-specialists/opportunities?live-results=true')
+        data = json.loads(res.get_data(as_text=True))
+
+        assert set(data.keys()) == {'results', 'summary', 'categories'}
+
+        for k, v in data.items():
+            assert set(v.keys()) == {'selector', 'html'}
+
+            # We want to enforce using css IDs to describe the nodes which should be replaced.
+            assert v['selector'].startswith('#')
+
+    @pytest.mark.parametrize('query_string, urls',
+                             (('', {'search/briefs.html'}),
+                              ('?live-results=true', {"search/_results_wrapper.html",
+                                                      "search/_categories_wrapper.html",
+                                                      "search/_summary.html",
+                                                      })))
+    @mock.patch('app.main.views.marketplace.render_template', autospec=True)
+    def test_base_page_renders_search_services(self, render_template_patch, query_string, urls):
+        render_template_patch.return_value = '<p>some html</p>'
+
+        self.client.get('/digital-outcomes-and-specialists/opportunities{}'.format(query_string))
+
+        assert urls == set(x[0][0] for x in render_template_patch.call_args_list)
+
+    def test_form_has_js_hidden_filter_button_chris(self):
```

Comment:
`_chris` ?

Reply:
😁

Reply:
oh snap! I was using my name to target specific tests with the `-k` flag! Forgot to remove.....